python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -477,40 +500,43 @@ def run(self, *, exposures, id_generator=None):
                 Reference catalog stars matches used in the photometric fit.
                 (`list` [`lsst.afw.table.ReferenceMatch`] or `lsst.afw.table.BaseCatalog`)
         """
+        if result is None:
+            result = pipeBase.Struct()
         if id_generator is None:
             id_generator = lsst.meas.base.IdGenerator()
 
-        exposure = self._handle_snaps(exposures)
+        result.exposure = self._handle_snaps(exposures)
 
         # TODO remove on DM-43083: work around the fact that we don't want
         # to run streak detection in this task in production.
-        exposure.mask.addMaskPlane("STREAK")
+        result.exposure.mask.addMaskPlane("STREAK")
 
-        psf_stars, background, candidates = self._compute_psf(exposure)
+        result.psf_stars_footprints, result.background, candidates = self._compute_psf(result.exposure)
+        result.psf_stars = result.psf_stars_footprints.asAstropy()
 
-        self._measure_aperture_correction(exposure, psf_stars)
+        self._measure_aperture_correction(result.exposure, result.psf_stars)
 
-        stars = self._find_stars(exposure, background, id_generator)
-        self._match_psf_stars(psf_stars, stars)
+        result.stars_footprints = self._find_stars(result.exposure, result.background, id_generator)
+        self._match_psf_stars(result.psf_stars_footprints, result.stars_footprints)
+        result.stars = result.stars_footprints.asAstropy()
 
-        astrometry_matches, astrometry_meta = self._fit_astrometry(exposure, stars)
-        stars, photometry_matches, photometry_meta, photo_calib = self._fit_photometry(exposure, stars)
+        astrometry_matches, astrometry_meta = self._fit_astrometry(result.exposure, result.stars_footprints)
+        if self.config.optional_outputs:
+            result.astrometry_matches = lsst.meas.astrom.denormalizeMatches(astrometry_matches,
+                                                                            astrometry_meta)
+
+        result.stars_footprints, photometry_matches, \
+            photometry_meta, result.applied_photo_calib = self._fit_photometry(result.exposure,
+                                                                               result.stars_footprints)
+        # fit_photometry returns a new catalog, so we need a new astropy table view.
+        result.stars = result.stars_footprints.asAstropy()
+        if self.config.optional_outputs:
+            result.photometry_matches = lsst.meas.astrom.denormalizeMatches(photometry_matches,
+                                                                            photometry_meta)
 
-        self._summarize(exposure, stars, background)
+        self._summarize(result.exposure, result.stars_footprints, result.background)
```

Comment:
Why is this using `result.stars_footprints` and not `result.stars` ?

Reply:
`ComputeExposureSummaryStatsTask` takes a SourceCatalog. `result.stars` is just for the output. Eventually we'll start converting downstream tasks (like ipdiffim, maybe?) to take `AstropyArrow` input catalogs, but that's a long ways off.

Reply:
Ah, so `stars_footprints` and `stars` has the same information but in different formats? Cute, OK, carry on.

Reply:
See the Connections here: https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/calibrateImage.py#L87

Eventually, the goal will be fore the `stars_footprints` to only contain the footprints and none of the catalog information, but that's a *long* ways off.